x/rollapp/keeper/block_height_to_finalization_queue.go

```diff
@@ -60,18 +62,21 @@ func (k Keeper) FinalizeRollappStates(ctx sdk.Context) {
 	}
 	// check to see if there are pending  states to be finalized
 	finalizationHeight := uint64(ctx.BlockHeight() - int64(k.DisputePeriodInBlocks(ctx)))
-	pendingFinalizationQueue := k.GetAllFinalizationQueueUntilHeightInclusive(ctx, finalizationHeight)
+	queue, err := k.GetFinalizationQueueUntilHeightInclusive(ctx, finalizationHeight)
+	if err != nil {
```

Comment:
i believe we cannot completely get rid of all panic in this end blocker, otherwise we would need to add some complex recovery mechanism. imagine the following:
1. we start processing some queue and it has 10 states to finalize
2. we finalize 5 states and fail on 6th (we run finalization in the cached ctx)
3. now we need to store this queue to the state
4. `SaveQueue` fails

we can't just log this and go on since the queue has already finalized states and thus will panic on the next iteration. and we can't make the finalization atomic since we support "partial" finalization (see no.2). to handle this we need to deeply refactor the method.

good news is that all panics there are encoding-related (i double checked!), so they should never happen in practice. i don't think we need to spend a lot of time for refactoring just to avoid these panics since they are not likely to happen (plus we've already had possible encoding panics in this method).

i added detailed reasoning to the lines that may panic.